datalake/src/main/java/com/sequenceiq/datalake/flow/dr/backup/handler/DatalakeFullBackupWaitHandler.java

```diff
@@ -33,7 +33,7 @@ public class DatalakeFullBackupWaitHandler extends ExceptionCatcherEventHandler<
     @Value("${sdx.stack.backup.status.sleeptime_sec:20}")
     private int sleepTimeInSec;
 
-    @Value("${sdx.stack.backup.status.duration_min:240}")
+    @Value("${sdx.stack.backup.status.duration_min:720}")
```

Comment:
Can we change it to 14 hours? That would address the longest backup we ever had.
This should be good till we have a logic to dynamically decide on the duration of the backup.

Reply:
Do we have any statistics how many customers need this long polling or how frequently do we timeout with the 240 minutes?
My guess it's a handful, and if thats true we are optimising for the wrong crowd.
I'm still worried that a thread will be blocked for 12-14 hours and this operation cannot be cancelled, so the cluster can't be stopped/scaled etc.

What about an entitlement that you can grant to certain customers where they have these extreme conditions.

Reply:
Or even better, we give some control to the customer by making the back flow cancellable.

Here you can see an example implementation for diagnostic bundles:
https://github.com/hortonworks/cloudbreak/blob/f7e3898f7980fa38330817a758873c15894b2a31/core/src/main/java/com/sequenceiq/cloudbreak/controller/v4/DiagnosticsV4Controller.java#L77

Reply:
Why is this resolved? What is the decision here?

Reply:
You are correct. That is not a majority case. There are a handful of customers who have huge metadata where the backup is there taking longer than 10 hours. 

In the last 2.5 months, there were 23 backups that took longer than two hours but none of them took more than 32 hours. I know cases where it took 10-12 hours in the past.
Below board will provide more information
[](https://eng-metrics.cdsw.edh.cloudera.com/arc/apps/builder/2521)

I see two suggestions
1. Use an entitlement to limit the cases where the timeout is longer.
2. Expose an option to cancel the operation so that customers could cancel backup/restore operation if they are impacted by a longer timeout interval. 

Here is the issue with the cancel operation. We can't really cancel the background backup/restore operations that are in progress in the SDX cluster. We do have a CDP-CLI command to cancel the backup/restore operations making the operation canceled in the control plane but the operations in the SDX cluster will still continue to run. This will cause issues if the customers try to retry again.

Reply:
Understood. I'm aware that it's hard to to cancellation properly, I'm fine with the entitlement.